ibm/service/iamaccessgroup/resource_ibm_iam_access_group_account_settings.go

```diff
@@ -0,0 +1,81 @@
+package iamaccessgroup
+
+import (
+	"context"
+	"fmt"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceIBMIAMAccessGroupAccountSettings() *schema.Resource {
+	return &schema.Resource{
+		ReadContext:   resourceIBMIAMAccessGroupAccountSettingGet,
+		UpdateContext: resourceIBMIAMAccessGroupAccountSettingSet,
+		CreateContext: resourceIBMIAMAccessGroupAccountSettingSet,
+		DeleteContext: resourceIBMIAMAccessGroupAccountSettingUnSet,
+		Importer:      &schema.ResourceImporter{},
+		Schema: map[string]*schema.Schema{
+			"public_access_enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Flag to enable/disable public access groups",
+			},
+		},
+	}
+}
+
+func resourceIBMIAMAccessGroupAccountSettingGet(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	iamAccessGroupsClient, err := meta.(conns.ClientSession).IAMAccessGroupsV2()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	userDetails, err := meta.(conns.ClientSession).BluemixUserDetails()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	getAccessGroupOptions := iamAccessGroupsClient.NewGetAccountSettingsOptions(userDetails.UserAccount)
+	accountSetting, detailedResponse, err := iamAccessGroupsClient.GetAccountSettings(getAccessGroupOptions)
+	if err != nil || accountSetting == nil {
+		if detailedResponse != nil && detailedResponse.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(fmt.Errorf("[ERROR] Error retrieving access group: %s. API Response: %s", err, detailedResponse))
+	}
+	d.SetId(*accountSetting.AccountID)
```

Comment:
account_id is not defined in the schema

Reply:
@hkantare account id is not directly defined in the setting template. so we picked up using "id" identifier for account id. do we need to add two variables in final state file as below:
``` {
          "schema_version": 0,
          "attributes": {
            "account_id": "b26bbfa828414cd7a2cb2eacd4711d61",
            "id": "b26bbfa828414cd7a2cb2eacd4711d61",
            "public_access_enabled": false
          },
          "sensitive_attributes": [],
          "private": "bnVsbA=="
        }
```


Please add more details if we are expecting something else.

Reply:
Then you should define in the schema right
```
Schema: map[string]*schema.Schema{
			"public_access_enabled": {
				Type:        schema.TypeBool,
				Optional:    true,
				Description: "Flag to enable/disable public access groups",
			},
			"account_id": {
				Type:        schema.TypeString,
				Computed:    true,
				Description: "xxxxxx",
			},
		},
		````